src/Units/UnitModelScheme.java

```diff
@@ -0,0 +1,62 @@
+package Units;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import Terrain.TerrainType;
+import Units.UnitModel.UnitRoleEnum;
+
+/**
+ * Provides the primary interface for the game to get info about game-specific units.
+ */
+public abstract class UnitModelScheme implements Serializable
+{
+  private static final long serialVersionUID = 1L;
+  
+  /** Indicates whether, for the given map/game, all the units can be instantiated */
+  public boolean schemeValid = true;
+
+  /**
+   * When the UI wants to show a unit type, it should use this string.
+   * Typically the name of whatever passes for infantry, but can be anything.
+   */
+  public abstract String getIconicUnitName();
```

Comment:
This function is overridden, but never actually used anywhere.

Reply:
I originally was using it to set the unit used for picking your palette/faction choice, but then I realized how awful an idea it was since the non-DoR factions would all default to the same bike sprite.

Reply:
I mean, if you are playing with the DoR unit set, then non-DoR factions _would_ have bikes. I can go either way on this.

Reply:
That's true, but then you can't tell which faction is which... which is still important, since your infantry are still different.
I guess we could figure out whether the given faction has its own sprite for the iconic unit, then pick INFANTRY as a default if it doesn't?

Reply:
Clearly, the correct solution is to check if the iconic type is available, then iterate through the UnitRoleEnum looking for a unique unit image.
...I was initially joking, but that actually doesn't sound half-bad.

Reply:
Uhhh. I mean. You do you, man. I feel like the UnitScheme might be a more definitive thing than the faction. Don't bikes have to be available to the non-DoR factions anyway?

Reply:
All of what you said is true... but it's extremely un-useful to just see row on row of identical bikes, with no indication of what factions they actually are.
When you're picking the faction, it kinda makes sense to show imagery based on *that* faction rather than falling through to whatever their default is.

I think I'll play around with this a bit and see what happens..

Reply:
If you want a better indication of what the factions actually are, then maybe we just need an overlay with the faction/color name as you scroll around. That way you always know what combination you're getting.

Reply:
I think that's a good idea, but I also think it's fair to only show a bike if the faction in question has a custom bike.
If the penguins don't have penguins on bikes, then maybe just show a normal penguin instead.
That way you don't have to manually check each faction with a duplicate/default unit.

Reply:
Every faction is supposed to have custom infantry though, right? That's how they've always been distinguished. I'm actually not sure why we need to consider other unit types at all.

Reply:
Fair enough.